roc/version.go

```diff
@@ -72,3 +73,40 @@ func parseVersion(s string) SemanticVersion {
 
 	return v
 }
+
+func stringifyVersion(sv SemanticVersion) string {
+	return fmt.Sprintf("%d.%d.%d", sv.Major, sv.Minor, sv.Patch)
+}
+
+func checkVersionCompatibility(nativeVersion, bindingsVersion SemanticVersion) error {
+	if nativeVersion.Major != bindingsVersion.Major {
+		return fmt.Errorf(`
+Bindings are compatible with native C library only if Major versions are same
+Bindings version: %s
+C library version: %s`,
+			stringifyVersion(bindingsVersion), stringifyVersion(nativeVersion),
+		)
+	}
+
+	if nativeVersion.Minor > bindingsVersion.Minor {
+		return fmt.Errorf(`
+Bindings are compatible with native C library only if its Minor version is same or higher
+Bindings version: %s
+C library version: %s`,
+			stringifyVersion(bindingsVersion), stringifyVersion(nativeVersion),
+		)
+	}
+	return nil
+}
+
+var runVersionCheckOnce sync.Once
```

Comment:
nit: maybe rename to versionCheckOnce or versionCheckGuard? For clarify and for easier association with versionCheck() func.